src/Chapter1.hs

```diff
@@ -554,7 +562,11 @@ value after "=" where the condition is true.
 Casual reminder about adding top-level type signatures for all functions :)
 -}
 
-mid x y z = error "mid: not implemented!"
+mid :: Ord a => a -> a -> a -> a
+mid x y z
+    | ((max x y) <= z) = max x y
+    | ((max x z) <= y) = max x z
+    | ((max y z) <= x) = max y z
```

Comment:
As we mentioned, the compiler in Haskell is very attentive to the exhaustive pattern-matching. And here it would warn you that `Pattern matching is not exhaustive`, as the guards have quite complicated logic, and the compiler won't be able to prove that it covers all the cases.

Because of that, you will need to add another guard – `| otherwise = ...`, to tell the compiler, that your pattern matching is exhaustive 🙂